packages/react-components/lib/reports/log-management/custom-lookup-filter.tsx

```diff
@@ -22,72 +20,33 @@ const MenuProps = {
-  columnDef: Column<{ level: string; message: string; created: string; container: ContainerType }>;
-  onFilterChanged: (rowId: string, value: string | number | unknown) => void;
-}
-
-/**
- * Component created to override the default lookup filter.
- */
-
-export const CustomLookupFilterParser = (
-  props: CustomLookupFilterParserProps,
-): React.ReactElement => {
-  const { columnDef: columnDefRaw, onFilterChanged } = props;
-  /**
-   * The column type in the material-table library it's not working correctly. So as a workaround I'm assigning the types to the properties need it in this component.
-   */
-  const columnDef = columnDefRaw as {
-    tableData: EditCellColumnDef['tableData'];
-    filterOnItemSelect: unknown;
-    lookup: Record<string, string>;
-  };
-
-  return (
-    <CustomLookupFilter
-      tableId={columnDef.tableData.id}
-      filterValue={columnDef.tableData.filterValue}
-      filterOnItemSelect={columnDef.filterOnItemSelect}
-      onFilterChanged={onFilterChanged}
-      lookup={columnDef.lookup}
-    />
-  );
-};
-
 export interface CustomLookupFilterProps {
-  tableId: number;
-  filterValue: EditCellColumnDef['tableData']['filterValue'];
-  filterOnItemSelect: unknown;
-  onFilterChanged: (rowId: string, value: string | number | unknown) => void;
   lookup: Record<string, string>;
+  selectedFilter: string[];
+  setSelectedFilter: React.Dispatch<React.SetStateAction<string[]>>;
 }
 
 export const CustomLookupFilter = (props: CustomLookupFilterProps): React.ReactElement => {
-  const { tableId, filterValue, filterOnItemSelect, onFilterChanged, lookup } = props;
-  const [selectedFilter, setSelectedFilter] = React.useState(filterValue || []);
-  React.useEffect(() => {
-    setSelectedFilter(filterValue || []);
-  }, [filterValue]);
-
+  const { lookup, selectedFilter, setSelectedFilter } = props;
+  const [filterTracker, setFilterTracker] = React.useState<string[]>(selectedFilter);
   return (
-    <FormControl style={{ width: '100%' }}>
-      <InputLabel
-        htmlFor={'select-multiple-checkbox' + tableId}
-        style={{ marginTop: -16 }}
-      ></InputLabel>
+    <FormControl style={{ width: '100%', padding: '1rem' }}>
+      <InputLabel htmlFor={'select-multiple-checkbox'}>Level Filter</InputLabel>
       <Select
         multiple
-        value={selectedFilter}
+        value={filterTracker}
         onClose={() => {
-          if (filterOnItemSelect !== true) onFilterChanged(tableId.toString(), selectedFilter);
+          setSelectedFilter(filterTracker);
+          setFilterTracker(filterTracker);
         }}
         onChange={(event) => {
-          setSelectedFilter(event.target.value);
-          if (filterOnItemSelect === true) onFilterChanged(tableId.toString(), event.target.value);
+          if (typeof event.target.value === 'string') {
+            setFilterTracker([...filterTracker, event.target.value]);
+          } else {
+            setFilterTracker(event.target.value);
+          }
```

Comment:
These changes are not reported to the parent component, you need to call `setSelectedFilter`, but as mentioned above, `setSelectedFilter` is useless and buggy.

All these can be fixed by simply removing `filterTracker` state, and just use `selectedFilter` and a `onFilterChange` callback. The parent should be responsible for updating the filters.

Reply:
I think I remember why I did that. The `select` dropdown would close every time a new filter is selected which is bad for UX. Adding an extra `filterTracker` in onChange first before triggering `onFilterChange` in onClose prevents unnecessary rerenders but I guess that was not a good way to solve that issue. :sweat_smile: